besspin/cyberPhys/specs/state_machine_hacker_kiosk.txt

```diff
@@ -1,95 +1,90 @@
 @startuml
 
-state "Hacker Kiosk start up" as KioskUp
-KioskUp: boot OS
-KioskUp: connect to network
-KioskUp: start kiosk webpage
-[*] --> KioskUp: Kiosk powered up
+[*] --> Slide2_KioskSetup
+Slide2_KioskSetup : select Demo timer On/Off
 
-KioskUp --> KioskWait
+Slide2_KioskSetup --> Slide3: Timer choice selected
 
-' TODO: add a timeout to the wait?
-state "Wait for components" as KioskWait
-KioskWait: Wait for CMD_COMPONENT_READY(BESSPIN_TOOl)
-KioskWait: Wait for CMD_COMPONENT_READY(IGNITION)
+state "Slide 3 INTRODUCTION" as Slide3
+Slide3: Tx CMD_ACTIVE_SCENARIO(SCENARIO_BASELINE)
+Slide3: Tx CMD_HACK_ACTIVE(0x0)
+Slide3: TODO: restart inactive components? SCENARIO_SECURE_ECU and SCENARIO_SECURE_INFOTAINMENT?
 
-' TODO: handle error conditions
-KioskWait --> KioskReady: Success\nSend CMD_COMPONENT_READY(KIOSK)\nSend CMD_LED_SEQ(BOOT)
-state "Hacker kiosk ready" as KioskReady
-KioskReady: welcome page
 
+Slide3 --> Slide4: Click "Hack the car" button
 
-state "Baseline Scenario" as ScenarioBaseline {
-    state "Scenario intro" as ScenarioBaselineIntro
-    ' ScenarioBaselineIntro: Driver: get faimilar with the controls
-    ScenarioBaselineIntro: Hacker: intro to SSITH
-    KioskReady --> ScenarioBaselineIntro: hacker arrived\nSend CMD_ACTIVE_SCENARIO(BASELINE)
+Slide4 --> Slide5: Click "Exploit OTA update" button
 
-    state "Explain OTA server hack" as ScenarioBaselineExplainOtaHack {
-        ' ScenarioBaselineExplainOtaHack: Driver: drives to te airport
-        ScenarioBaselineExplainOtaHack: Hacker: explain OTA hack
-    }
+Slide5 --> Slide6_A: Click "Hack infotainment" button
 
-    ScenarioBaselineIntro --> ScenarioBaselineExplainOtaHack: Hacker clicks though the intro slides
-    ScenarioBaselineExplainOtaHack --> ScenarioBaselineHackedOTA: Hacker clicks in "Hack OTA update server"\nSend CMD_LED_SEQ(OTA_HACK)
-    state "OTA server hacked" as ScenarioBaselineHackedOTA
-    ScenarioBaselineHackedOTA: signing key erased
+state "Slide 6a HACK OTA SERVER" as Slide6_A
+Slide6_A: Call "OtaHack.hackServer()"
 
-    state "Hack infotainment" as ScenarioBaselineHackInfotainment
-    ScenarioBaselineHackInfotainment: show available infotainment hacks
+Slide6_A --> Slide6_B: "OtaHack.hackServer()" returned True
```

Comment:
What happens if it didn't return `True`? Do we have (or need) an escape hatch if the hack doesn't work/something goes wrong?

Reply:
Great question. The UI doesn't really have a way of showing failures - so we can at best log it/show it in the console. What do you think?

Reply:
I think that's fine as far as it goes... but it's not clear what the UI should _do_ in the event of a failure. We can't really just move on to the infotainment hack screen.